osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -417,16 +426,23 @@ private void removeBeatmapSet(Guid beatmapSetID)
             }
         }
 
-        public void UpdateBeatmapSet(BeatmapSetInfo beatmapSet) => Schedule(() =>
+        public void UpdateBeatmapSet(BeatmapSetInfo beatmapSet)
         {
-            updateBeatmapSet(beatmapSet);
-            invalidateAfterChange();
-        });
+            beatmapSet = beatmapSet.Detach();
+
+            Schedule(() =>
+            {
+                updateBeatmapSet(beatmapSet);
+                invalidateAfterChange();
+            });
+        }
 
         private void updateBeatmapSet(BeatmapSetInfo beatmapSet)
         {
+            beatmapSet = beatmapSet.Detach();
```

Comment:
This is also a very scary looking change to drop sans context (means all `CarouselBeatmapSet`s will receive detached models), but looking at `master` it looks to actually _not_ be a behavioural change since that is the case there already, and this is presumably added because of the unexplained refetch in `processBeatmapChanges()`...?

Reply:
I added this because we're already detaching anyway (for the detached list) so it makes sense to query from the detached model to save extra realm round-trips. Nothing more than that.